### PR TITLE
Add support for adxl380 and adxl382

### PR DESCRIPTION
**Pull request by rbudai98**
# Description

Add py-adi support for adxl380 and adxl382

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

**Test Configuration**:
* Hardware:  RPi4 + eval board adxl38x
* OS: Kuiper and liunx driver

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: supported_parts.md [hunk 1]
```diff
@@ -160,6 +160,7 @@
 - ADXL314
 - ADXL345
 - ADXL355
+- ADXL380
```
Comment (buha) [inline supported_parts.md:163]: shouldn't this contain ADXL382 as well ?

File: test/emu/hardware_map.yml [hunk 2]
```diff
@@ -327,6 +327,14 @@ adxl345:
       - filename: adxl345.xml
       - data_devices:
           - iio:device0
+adxl380:
```
Comment (buha) [inline test/emu/hardware_map.yml:330]: what about adxl382 ?

File: test/emu/devices/adxl380.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+<!ELEMENT buffer-attribute EMPTY>
+<!ATTLIST context name CDATA #REQUIRED description CDATA #IMPLIED>
+<!ATTLIST context-attribute name CDATA #REQUIRED value CDATA #REQUIRED>
+<!ATTLIST device id CDATA #REQUIRED name CDATA #IMPLIED>
+<!ATTLIST channel id CDATA #REQUIRED type (input|output) #REQUIRED name CDATA #IMPLIED>
+<!ATTLIST scan-element index CDATA #REQUIRED format CDATA #REQUIRED scale CDATA #IMPLIED>
+<!ATTLIST attribute name CDATA #REQUIRED filename CDATA #IMPLIED value CDATA #IMPLIED>
+<!ATTLIST debug-attribute name CDATA #REQUIRED value CDATA #IMPLIED>
+<!ATTLIST buffer-attribute name CDATA #REQUIRED value CDATA #IMPLIED>]>
+<context name="network" description="10.42.0.83 Linux analog 6.1.54-v7l+ #9 SMP Wed Oct 30 07:08:07 GMT 2024 armv7l" >
```
Comment (buha) [inline test/emu/devices/adxl380.xml:19]: delete the ip or replace it with 127.0.0.1

File: supported_parts.md (hunk 1 above)
Comment (rbudai98) [reply supported_parts.md:163]: added adxl382 as well

File: test/emu/devices/adxl380.xml (hunk 3 above)
Comment (rbudai98) [reply test/emu/devices/adxl380.xml]: removed

**Comment by rbudai98** [comment]
v2:
* applied PR comments

**Review by tfcollins — CHANGES_REQUESTED** [review]

File: adi/adxl380.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+                        print("Found device = " + i + ". Will use this device instead.")
+                        break
+                if self._ctrl is None:
+                    raise Exception("No compatible device found")
+
+            self.accel_x = self._channel(self._ctrl, "accel_x")
+            self.accel_y = self._channel(self._ctrl, "accel_y")
+            self.accel_z = self._channel(self._ctrl, "accel_z")
+            self.temp = self._tempchannel(self._ctrl, "temp")
+            self._rxadc = self._ctx.find_device("adxl380")
```
[the remark this answers is not in the document]
Comment (rbudai98) [reply adi/adxl380.py]: Removed, it was an extra line from initial versioning

**Pull request merged**